### ISSUE TITLE
fix: Make sure to register a handler via on<$eventType>((event, emit)

### ISSUE THREAD

**Issue by imcongpham** [felangel/bloc#3577]
**Description**
Hi I'm having an issue saying **Make sure to register a handler via on<$eventType>((event, emit)** . At first, I thought I was missing something from my source code but even I'm using the example from https://pub.dev/packages/flutter_bloc/example the issue is still showing...

full log: 

```
Error: 'ngthrow' isn't a type.
../…/src/bloc.dart:90
            ngthrow StateError(
            ^^^^^^^
: Error: Expected ';' after this.
../…/src/bloc.dart:90
            ngthrow StateError(
                    ^^^^^^^^^^
: Error: Expected ')' before this.
../…/src/bloc.dart:92
              '''Make sure to register a handler via on<$eventType>((event, emit) {...})''',
```

**Steps To Reproduce**

1. create new flutter app
2. add flutter doc dependency flutter_bloc: ^8.1.1
3. copy whole this example from main.dart
4. run

**Expected Behavior**
The new app should run

**Screenshots**

**Additional Context**

OS: MacOS Montery
VS Code specs: 

```
Version: 1.72.0 (Universal)
Commit: 64bbfbf67ada9953918d72e1df2f4d8e537d340e
Date: 2022-10-04T23:21:58.256Z (1 wk ago)
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: No
```

my pubspec.yaml

```
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  http: ^0.12.2
  flutter_bloc: ^8.1.1

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.1

dev_dependencies:
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter

```

**Comment by Luci3b** [comment]
I followed the steps and I couldn't reproduce this bug on my computer.

Flutter --version
```bash
Flutter 3.3.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision eb6d86ee27 (11 days ago) • 2022-10-04 22:31:45 -0700
Engine • revision c08d7d5efc
Tools • Dart 2.18.2 • DevTools 2.15.0
```
Flutter_bloc package version: 8.1.1

Please share more details about your flutter version and your code.

**Comment by felangel** [comment]
@imcongpham can you please provide a link to a minimal reproduction sample? Thanks!

**Comment by imcongpham** [comment]
> I followed the steps and I couldn't reproduce this bug on my computer.
> 
> Flutter --version
> 
> ```shell
> Flutter 3.3.4 • channel stable • https://github.com/flutter/flutter.git
> Framework • revision eb6d86ee27 (11 days ago) • 2022-10-04 22:31:45 -0700
> Engine • revision c08d7d5efc
> Tools • Dart 2.18.2 • DevTools 2.15.0
> ```
> 
> Flutter_bloc package version: 8.1.1
> 
> Please share more details about your flutter version and your code.

I have same flutter version

```shell
Flutter 3.3.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision eb6d86ee27 (13 days ago) • 2022-10-04 22:31:45 -0700
Engine • revision c08d7d5efc
Tools • Dart 2.18.2 • DevTools 2.15.0
```
my temporary fix is now rollback to 7.3.3 version and keep using mapEventToState...

@felangel I have posted the steps to reproduce, but seems like only me seeing it so far... I'll wait for some time and retry the build again.

**Comment by felangel** [comment]
> > I followed the steps and I couldn't reproduce this bug on my computer.
> > Flutter --version
> > ```shell
> > Flutter 3.3.4 • channel stable • https://github.com/flutter/flutter.git
> > Framework • revision eb6d86ee27 (11 days ago) • 2022-10-04 22:31:45 -0700
> > Engine • revision c08d7d5efc
> > Tools • Dart 2.18.2 • DevTools 2.15.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Flutter_bloc package version: 8.1.1
> > Please share more details about your flutter version and your code.
> 
> I have same flutter version
> 
> ```shell
> Flutter 3.3.4 • channel stable • https://github.com/flutter/flutter.git
> Framework • revision eb6d86ee27 (13 days ago) • 2022-10-04 22:31:45 -0700
> Engine • revision c08d7d5efc
> Tools • Dart 2.18.2 • DevTools 2.15.0
> ```
> 
> my temporary fix is now rollback to 7.3.3 version and keep using mapEventToState...
> 
> @felangel I have posted the steps to reproduce, but seems like only me seeing it so far... I'll wait for some time and retry the build again.

Can you please provide a link to a minimal reproduction sample? I’d also recommend clearing your pub cache and/or reinstalling Flutter to see if that resolves the issue.

**Comment by felangel** [comment]
Closing for now since there are no actionable next steps or reproduction sample. Feel free to provide a link to a reproduction sample or provide additional context if the suggestions don't help, thanks!
